### PR TITLE
Respect frame on plugin thrown errors when pos is also given

### PR DESCRIPTION
**Pull request by Andarist**
Would love to add a test for this - hadn't yet chance to search for similar things being tested though.

### CODE REVIEW

File: src/utils/transform.ts [hunk 1]
```diff
@@ -40,12 +40,15 @@ function augmentCodeLocation<T extends RollupError | RollupWarning>({
...
 		} else {
 			object.pos = <any>pos;
 			const { line, column } = locate(source, pos, { offsetLine: 1 });
 			object.loc = { file: id, line, column };
-			object.frame = getCodeFrame(source, line, column);
+		}
+
+		if (object.frame === undefined) {
+			const { line, column } = object.loc;
+			object.frame = object.frame || getCodeFrame(source, line, column);
```
Comment (guybedford) [inline src/utils/transform.ts]: This can just be `object.frame = getCodeFrame(` then?

**Comment by Andarist** [comment]
> This can just be object.frame = getCodeFrame( then?

Obviously, this was last second refactor 😅

> How about a simple API-based test in test/hooks?

Gonna prepare one in following days. Newborn in the house 😉

**Comment by Andarist** [comment]
@guybedford added a CLI test

File: test/cli/samples/custom-frame/rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	},
+	plugins: [
+		{
+			transform() {
+				const err = new Error('My error.');
+				err.loc = {
+					line: 1,
+					column: 5,
+				};
+				err.frame = 'custom code frame';
```
Comment (lukastaegert) [inline test/cli/samples/custom-frame/rollup.config.js]: It puzzled me a little why your test was green even without your changes to the code and it seems this test does not even trigger your code changes as no code position is provided as the second argument. This does not mean that I would change this test—it definitely tests something important—but that there should be a second test where `this.error` is called with a second argument, either a `{line, column}` object or a plain number (or just call it twice :) you might also just put all three invocations into one test).

**Comment by lukastaegert** [comment]
> Gonna prepare one in following days. Newborn in the house 😉

And congratulations!

**Comment by Andarist** [comment]
@lukastaegert I've made suggested amends. Would also appreciate you taking a look at [this](https://github.com/rollup/rollup/issues/2307#issuecomment-401475658)

**Pull request merged**